iothub/device/src/Transport/AmqpIoT/AmqpIoTSession.cs

```diff
@@ -202,7 +206,7 @@ TimeSpan timeout
             if (Logging.IsEnabled) Logging.Enter(typeof(AmqpIoTSession), deviceIdentity, $"{nameof(OpenSendingAmqpLinkAsync)}");
             AmqpLinkSettings amqpLinkSettings = new AmqpLinkSettings
             {
-                LinkName = CommonResources.GetNewStringGuid(linkSuffix),
+                LinkName = linkSuffix,
```

Comment:
From the AMQP spec:

>2.6.1 Naming A Link
>Links are named so that they may be recovered when communication is interrupted. Link names MUST uniquely identify the link amongst all links of the same direction between the two participating containers. Link names are only used when attaching a link, so they may be arbitrarily long without a significant penalty.

Why isn't this unique anymore? Looks like breaking the standard.

Reply:
It is unique. device and module id will be append into link suffix.

Reply:
That doesn't make it unique: you can have the link detached and the session closing on this side while the service's connection to the broker still has the same name.
Either way: what are you gaining by removing the unique GUID?

Reply:
That's a great example why it should be the same name. It means server and client is now out of sync. It will trigger connection closed and reconnect.

Reply:
As per the discussion above, it looks resolved. Please reopen if not so.

Reply:
@prmathur-microsoft I think changing the protocol implementation should get some sign-off from the service team to make sure we’re not going to corrupt data. 
My interpretation is that all links should be unique. We need to consider that one AMQP broker can in theory get connections from multiple clients (sometimes the same client connecting multiple times.